### PR TITLE
make it work when the package is not loaded

### PR DESCRIPTION
**Pull request by dfalbel**
When the package is not loaded into the environment I get the following error:

``` r
iter <- flowery::generator({
  for (x in letters) yield(x)
})

flowery::iterate(for(x in iter){
  print(x)
})
#> Error in is_done(`_for_iter_2`): could not find function "is_done"
```

<sup>Created on 2019-04-16 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

This PR adds `flowery::` so `iterate` works when the package is not loaded.

### CODE REVIEW

**Comment by lionel-** [comment]
Thanks! Could you also add a unit test please? Something like executing the generator in a local environment inheriting from base.

**Comment by dfalbel** [comment]
Just added the unit test. Also tested on master and it errors so it's probably good to go

**Comment by lionel-** [comment]
Thanks!

File: tests/testthat/test-iter-for.R [hunk 1]
```diff
@@ -54,3 +54,15 @@ test_that("iterating with a done iterator cause an error", {
   iterate(for (x in iter) x)
   expect_error(iterate(for (x in iter) x), "is done")
 })
+
+test_that("iterating works when flowery is not loaded", {
+  new_env <- new.env(parent = baseenv())
+
+  eval(
```
Comment (lionel-) [inline tests/testthat/test-iter-for.R]: We usually use `evalq()` for this by the way.

**Pull request merged**